lib/dao.js

```diff
@@ -2638,8 +2638,12 @@ DataAccessObject.replaceById = function(id, data, options, cb) {
   assert(typeof cb === 'function', 'The cb argument must be a function');
 
   var connector = this.getConnector();
-  assert(typeof connector.replaceById === 'function',
-          'replaceById() must be implemented by the connector');
+
+  if (typeof connector.replaceById !== 'function') {
+    var err =  new Error('there is no support for replaceById() in the connector.' +
+      'Connector needs to add support for `replaceById()`');
+    return cb(err);
+  }
```

Comment:
> TLDR: We should make all the other asserts operate in the same fashion as yours, but it's prob not worth the effort to go change all the other ones. We can discuss more during actual DAO refactorings.

Actually I'm in doubt if the use of `assert` is right in DAO at all, any assertion failure dumps stack trace and causes the application failure which should not happen in DAO; I believe it is still worth fixing them, even before any refactoring which may happen in `loopback-datasource-juggler`. 

Here is another example: 
This assertion causues the following failure:
**Assertion:**
```
assert(arguments.length >= 1, 'At least one argument is required');
```
**Failure:**
```
assert.js:89
  throw new assert.AssertionError({
  ^
AssertionError: At least one argument is required
    at Function.findOrCreate (/Users/jafaria/workspace/replaceDemo/node_modules/loopback-datasource-juggler/lib/dao.js:836:3)
    at /Users/jafaria/workspace/replaceDemo/server/boot/boot.js:62:17
    at /Users/jafaria/workspace/replaceDemo/node_modules/loopback-connector-db2/node_modules/async/lib/async.js:52:16
    at done (/Users/jafaria/workspace/replaceDemo/node_modules/loopback-connector-db2/node_modules/async/lib/async.js:246:17)
    at /Users/jafaria/workspace/replaceDemo/node_modules/loopback-connector-db2/node_modules/async/lib/async.js:44:16
    at /Users/jafaria/workspace/replaceDemo/node_modules/loopback-connector-db2/node_modules/loopback-connector/lib/sql.js:1348:9
    at /Users/jafaria/workspace/replaceDemo/node_modules/loopback-connector-db2/node_modules/async/lib/async.js:726:13
    at /Users/jafaria/workspace/replaceDemo/node_modules/loopback-connector-db2/node_modules/async/lib/async.js:52:16
    at /Users/jafaria/workspace/replaceDemo/node_modules/loopback-connector-db2/node_modules/async/lib/async.js:269:32
    at /Users/jafaria/workspace/replaceDemo/node_modules/loopback-connector-db2/node_modules/async/lib/async.js:44:16
```
I can take care of all them.